Sources/_StringProcessing/ByteCodeGen.swift

```diff
@@ -1324,6 +1505,51 @@ extension DSLTree.Node {
     default: return false
     }
   }
+
+  /// A Boolean value indicating whether this node reverses the match position
+  /// on a successful match.
+  ///
+  /// For example, an alternation like `(a|b|c)` always advances the position
+  /// by a character, but `(a|b|)` has an empty branch, which matches without
+  /// advancing.
+  var guaranteesBackwardProgress: Bool {
```

Comment:
Is the logic in here different than guaranteeing forward progress?

Reply:
Doesn't look like it, I'll remove it